src/transformers/modeling_longformer.py

```diff
@@ -287,178 +471,11 @@ def _sliding_chunks_matmul_attn_probs_value(
         )
         chunked_value = padded_value.as_strided(size=chunked_value_size, stride=chunked_value_stride)
 
-        chunked_attn_probs = self._pad_by_window_overlap_except_last_row(chunked_attn_probs)
+        chunked_attn_probs = self._pad_and_diagonalize(chunked_attn_probs)
 
         context = torch.einsum("bcwd,bcdh->bcwh", (chunked_attn_probs, chunked_value))
         return context.view(batch_size, num_heads, seq_len, head_dim).transpose(1, 2)
 
-    def forward(
```

Comment:
just moved the forward up, because I think that's the first function one should look at.

Reply:
next time consider never resolving this and putting it higher so that reviewer sees it and understands. Took me  30s to get to the bottom.